plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfigurationProvider.java

```diff
@@ -109,6 +111,16 @@ public void updateConfiguration(Configuration configuration, HdfsContext context
             hasher.putString(role, UTF_8);
         });
 
+        mapping.getKmsKeyId().ifPresent(key -> {
+            configuration.set(S3_KMS_KEY_ID, key);
+            hasher.putString(key, UTF_8);
+        });
+
+        mapping.getSseKmsKeyId().ifPresent(key -> {
+            configuration.set(S3_SSE_KMS_KEY_ID, key);
+            hasher.putString(key, UTF_8);
```

Comment:
I think we should have `hasher.putString(S3_SSE_KMS_KEY_ID +":" + key, UTF_8)` or something else to distinguish case where  `S3_KMS_KEY_ID=xyz` from `S3_SSE_KMS_KEY_ID=xyz`

Reply:
Good catch.